src/core/server/saved_objects/migrationsv2/integration_tests/multiple_es_nodes.test.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import Fs from 'fs';
+import Util from 'util';
+import glob from 'glob';
+import { kibanaServerTestUser } from '@kbn/test';
+import { kibanaPackageJson as pkg } from '@kbn/utils';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import type { ElasticsearchClient } from '../../../elasticsearch';
+import { Root } from '../../../root';
+
+const LOG_FILE_PREFIX = 'migration_test_multiple_es_nodes';
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+
+async function removeLogFile() {
+  glob(Path.join(__dirname, `${LOG_FILE_PREFIX}_*.log`), (err, files) => {
+    files.forEach(async (file) => {
+      // ignore errors if it doesn't exist
+      await asyncUnlink(file).catch(() => void 0);
+    });
+  });
+}
+
+function extractSortNumberFromId(id: string): number {
+  const parsedId = parseInt(id.split(':')[1], 10); // "foo:123" -> 123
+  if (isNaN(parsedId)) {
+    throw new Error(`Failed to parse Saved Object ID [${id}]. Result is NaN`);
+  }
+  return parsedId;
+}
+
+async function fetchDocs(esClient: ElasticsearchClient, index: string, type: string) {
+  const { body } = await esClient.search<any>({
+    index,
+    size: 10000,
+    body: {
+      query: {
+        bool: {
+          should: [
+            {
+              term: { type },
+            },
+          ],
+        },
+      },
+    },
+  });
+
+  return body.hits.hits
+    .map((h) => ({
+      ...h._source,
+      id: h._id,
+    }))
+    .sort((a, b) => extractSortNumberFromId(a.id) - extractSortNumberFromId(b.id));
+}
+
+interface RootConfig {
+  logFileName: string;
+  hosts: string[];
+}
+
+function createRoot({ logFileName, hosts }: RootConfig) {
+  return kbnTestServer.createRoot({
```

Comment:
Could you either use the `oss` dist or disable `reporting` plugin? See https://github.com/elastic/kibana/issues/102919

Reply:
We'll be removing the OSS dist eventually - once the tests are migrated out.  cc @tylersmalley

Reply:
@mshustov Ah, thanks for pointing that out. Yes, will disable reporting.

> We'll be removing the OSS dist eventually

++ Yep, I was avoiding using `oss` for this reason.

Reply:
After looking at this further, I am not specifying a license in these tests, which means it defaults to `basic`, which doesn't include reporting anyway. So I think we are okay here.

(I discovered this by setting `xpack.reporting.enabled: false`, which threw a config validation error since the plugin isn't included)